tests/utils_test.py

```diff
@@ -42,3 +46,46 @@ def test_get_loss_vector_with_custom_none_replacement():
         none_replacement=float("Inf"),
     )
     np.testing.assert_array_equal(loss_vector, np.array(expected))
+
+
+def test_mask_pareto_efficient():
+    evals = np.array(
+        [
+            [0.0, 1.0],
+            [1.1, 0.1],
+            [0.0, 1.0],
+            [1.0, 0.0],
+            [3.1, 1.1],
+            [0.1, 1.0],
+            [0.0, 1.1],
+            [-1.0, 2.0],
+        ]
+    )
+    pareto_efficient = mask_pareto_efficient(evals)
+    assert len(pareto_efficient) == evals.shape[0]
+    assert pareto_efficient[0]
+    assert not pareto_efficient[1]
+    assert pareto_efficient[2]
+    assert pareto_efficient[3]
+    assert not pareto_efficient[4]
+    assert not pareto_efficient[5]
+    assert not pareto_efficient[6]
+    assert pareto_efficient[7]
+
+
+def test_filter_pareto_efficient():
+    evals = [
+        Evaluation(configuration={"i": 0}, objectives={"loss": 0.0, "score": -1.0}),
+        Evaluation(configuration={"i": 1}, objectives={"loss": 1.1, "score": -0.1}),
+        Evaluation(configuration={"i": 2}, objectives={"loss": 0.0, "score": -1.0}),
+        Evaluation(configuration={"i": 3}, objectives={"loss": 1.0, "score": 0.0}),
+        Evaluation(configuration={"i": 4}, objectives={"loss": 3.1, "score": -1.1}),
+        Evaluation(configuration={"i": 5}, objectives={"loss": 0.1, "score": -1.0}),
+        Evaluation(configuration={"i": 6}, objectives={"loss": 0.0, "score": -1.1}),
+        Evaluation(configuration={"i": 7}, objectives={"loss": -1.0, "score": -2.0}),
+    ]
+    pareto_efficient = filter_pareto_efficient(
+        evals, [Objective("loss", False), Objective("score", True)]
+    )
+    assert len(pareto_efficient) == 4
+    assert set([0, 2, 3, 7]) == set([e.configuration["i"] for e in pareto_efficient])
```

Comment:
This is now duplicating test logic from `mask_pareto_efficient` more than it would need. Effectively, we just need to test here that the objective inversion and filtering of the list work. However, I found the overhead low enough and the benefit large enough that we get many edge cases here on this level as well to accept the duplication. Let me know what you think about this aspect.

Reply:
Agree, it's worth it!